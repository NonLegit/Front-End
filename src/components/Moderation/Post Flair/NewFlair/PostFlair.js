import axios from '../../../../services/instance';
/**
 * Add New Flair
 *
 * @property {boolean} url - url to send data on
 * @property {boolean} text - text flair
 * @property {boolean} backgroundColor - background color for flair
 * @property {boolean} textColor - text color
 *
 * @returns {object} provided response through backend
 */

const PostFlair = async (url, text, backgroundColor, textColor) => {
  // const [data, setData] = useState(null);
  // const [error, setError] = useState(null);
  await axios.post(
    url,
    JSON.stringify({
      text,
      backgroundColor,
      textColor,

    }),
  ).then((response) => {
    console.log(response.status);
    if (response?.status === 401) {
      window.location.pathname = 'login';
    }
    return response.status;
  }).catch((error) => {
    console.log(error);
    return false;
  });
};
export default PostFlair;
