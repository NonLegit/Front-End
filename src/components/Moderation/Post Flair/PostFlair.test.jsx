import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import PostFlair from './PostFlair';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<PostFlair />).toJSON();
  expect(tree).toMatchSnapshot();
});
