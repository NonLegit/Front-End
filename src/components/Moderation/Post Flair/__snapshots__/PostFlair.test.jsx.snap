// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="css-qu6rh0"
>
  <div
    className="css-e9p5qh"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ozqymz-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Add Flair
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="css-lm0ox3"
  >
    <h2
      className="css-udpovt"
    >
      Post flair management
      <button
        aria-label="Learn more"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-1f85vht-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-bk4aae-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </h2>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root css-kizp0d-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-11momgf-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-d8nq3b-MuiTableCell-root"
              scope="col"
            >
              POST FLAIR PREVIEW
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1jk6xnc-MuiTableCell-root"
              scope="col"
            >
              CSS CLASS
              <button
                aria-label="CSS classes determine the styling for flair in old Reddit"
                aria-labelledby={null}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-1f85vht-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element={true}
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z03sc4-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1jk6xnc-MuiTableCell-root"
              scope="col"
            >
              SETTINGS
              <button
                aria-label="Shows if flair is mod-only, user editable, allows text, or allows emojis"
                aria-labelledby={null}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-1f85vht-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element={true}
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z03sc4-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1vlysoi-MuiTableCell-root"
              scope="col"
            >
              FLAIR ID
              <button
                aria-label="Use this ID when configuring automod to assign flair"
                aria-labelledby={null}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-1f85vht-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element={true}
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z03sc4-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-1gbrtqw-MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </div>
</div>
`;
